Can't use Compress middleware in 4.0.0-beta.8

I tried upgrading a server with the following `App` initialization to actix-web 4.0.0-beta.8 (from 4.0.0-beta.5)

```rust
App::new()
    // .app_data()
    // .data() - multiple calls
    .wrap(middleware::Logger::default())
    .wrap(middleware::Compress::default())
    .configure(routes::config)
```

And the second-to-last line fails to compile with:

```
error[E0277]: the trait bound `actix_http::Error: From<actix_web::Error>` is not satisfied
  --> src/api/server.rs:49:14
   |
49 |             .wrap(middleware::Compress::default())
   |              ^^^^ the trait `From<actix_web::Error>` is not implemented for `actix_http::Error`
   |
   = help: the following implementations were found:
             <actix_http::Error as From<HandshakeError>>
             <actix_http::Error as From<Infallible>>
             <actix_http::Error as From<PayloadError>>
             <actix_http::Error as From<WsProtocolError>>
           and 3 others
   = note: required because of the requirements on the impl of `Into<actix_http::Error>` for `actix_web::Error`
   = note: required because of the requirements on the impl of `actix_web::dev::MessageBody` for `actix_http::encoding::Encoder<actix_web::middleware::logger::StreamLog<AnyBody>>`
   = note: 1 redundant requirements hidden
   = note: required because of the requirements on the impl of `actix_web::dev::MessageBody` for `ResponseBody<actix_http::encoding::Encoder<actix_web::middleware::logger::StreamLog<AnyBody>>>`
```

(there's also compiler errors with `routes::config` but I'm pretty sure that's an entirely separate problem)

I have made sure that there's only a single version of actix-http in the dependency tree:

```
$ cargo tree -i actix-http
actix-http v3.0.0-beta.9
├── actix-web v4.0.0-beta.8
│   ├── actix-cors v0.6.0-beta.2
│   │   └── my-crate v0.1.0 (path...)
│   ├── actix-multipart v0.4.0-beta.5
│   │   └── my-crate v0.1.0 (path...)
│   ├── actix-web-httpauth v0.6.0-beta.2
│   │   └── my-crate v0.1.0 (path...)
│   └── my-crate v0.1.0 (path...)
├── awc v3.0.0-beta.7
│   └── my-crate v0.1.0 (path...)
└── my-crate v0.1.0 (path...)
```

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

* Rust Version (I.e, output of `rustc -V`): rustc 1.54.0 (a178d0322 2021-07-26)
* Actix Web Version: 4.0.0-beta.8

--- a/actix-http/CHANGES.md
+++ b/actix-http/CHANGES.md
@@ -3,6 +3,7 @@
 ## Unreleased - 2021-xx-xx
 ### Changes
 * Minimum supported Rust version (MSRV) is now 1.51.
+* Remove `Into<Error>` bound on `Encoder` body types.
 
 ### Fixed
 * Remove slice creation pointing to potential uninitialized data on h1 encoder. [#2364]

--- a/actix-http/CHANGES.md
+++ b/actix-http/CHANGES.md
@@ -3,12 +3,13 @@
 ## Unreleased - 2021-xx-xx
 ### Changes
 * Minimum supported Rust version (MSRV) is now 1.51.
-* Remove `Into<Error>` bound on `Encoder` body types.
 
 ### Fixed
 * Remove slice creation pointing to potential uninitialized data on h1 encoder. [#2364]
+* Remove `Into<Error>` bound on `Encoder` body types. [#2375]
 
 [#2364]: https://github.com/actix/actix-web/pull/2364
+[#2375]: https://github.com/actix/actix-web/pull/2375
 
 
 ## 3.0.0-beta.8 - 2021-08-09

--- a/actix-http/src/body/message_body.rs
+++ b/actix-http/src/body/message_body.rs
@@ -11,8 +11,6 @@ use bytes::{Bytes, BytesMut};
 use futures_core::ready;
 use pin_project_lite::pin_project;
 
-use crate::error::Error;
-
 use super::BodySize;
 
 /// An interface for response bodies.
@@ -47,7 +45,6 @@ impl MessageBody for () {
 impl<B> MessageBody for Box<B>
 where
     B: MessageBody + Unpin,
-    B::Error: Into<Error>,
 {
     type Error = B::Error;
 
@@ -66,7 +63,6 @@ where
 impl<B> MessageBody for Pin<Box<B>>
 where
     B: MessageBody,
-    B::Error: Into<Error>,
 {
     type Error = B::Error;
 

--- a/actix-http/src/encoding/encoder.rs
+++ b/actix-http/src/encoding/encoder.rs
@@ -29,7 +29,7 @@ use crate::{
         header::{ContentEncoding, CONTENT_ENCODING},
         HeaderValue, StatusCode,
     },
-    Error, ResponseHead,
+    ResponseHead,
 };
 
 use super::Writer;
@@ -107,7 +107,6 @@ enum EncoderBody<B> {
 impl<B> MessageBody for EncoderBody<B>
 where
     B: MessageBody,
-    B::Error: Into<Error>,
 {
     type Error = EncoderError<B::Error>;
 
@@ -142,7 +141,6 @@ where
 impl<B> MessageBody for Encoder<B>
 where
     B: MessageBody,
-    B::Error: Into<Error>,
 {
     type Error = EncoderError<B::Error>;
 

--- a/actix-http/src/h1/utils.rs
+++ b/actix-http/src/h1/utils.rs
@@ -65,7 +65,9 @@ where
                     let next =
                         match this.body.as_mut().as_pin_mut().unwrap().poll_next(cx) {
                             Poll::Ready(Some(Ok(item))) => Poll::Ready(Some(item)),
-                            Poll::Ready(Some(Err(err))) => return Poll::Ready(Err(err.into())),
+                            Poll::Ready(Some(Err(err))) => {
+                                return Poll::Ready(Err(err.into()))
+                            }
                             Poll::Ready(None) => Poll::Ready(None),
                             Poll::Pending => Poll::Pending,
                         };

--- a/src/middleware/mod.rs
+++ b/src/middleware/mod.rs
@@ -19,3 +19,43 @@ mod compress;
 
 #[cfg(feature = "__compress")]
 pub use self::compress::Compress;
+
+#[cfg(test)]
+mod tests {
+    use crate::{http::StatusCode, App};
+
+    use super::*;
+
+    #[test]
+    fn common_combinations() {
+        // ensure there's no reason that the built-in middleware cannot compose
+
+        let _ = App::new()
+            .wrap(Compat::new(Logger::default()))
+            .wrap(Condition::new(true, DefaultHeaders::new()))
+            .wrap(DefaultHeaders::new().header("X-Test2", "X-Value2"))
+            .wrap(ErrorHandlers::new().handler(StatusCode::FORBIDDEN, |res| {
+                Ok(ErrorHandlerResponse::Response(res))
+            }))
+            .wrap(Logger::default())
+            .wrap(NormalizePath::new(TrailingSlash::Trim));
+
+        let _ = App::new()
+            .wrap(NormalizePath::new(TrailingSlash::Trim))
+            .wrap(Logger::default())
+            .wrap(ErrorHandlers::new().handler(StatusCode::FORBIDDEN, |res| {
+                Ok(ErrorHandlerResponse::Response(res))
+            }))
+            .wrap(DefaultHeaders::new().header("X-Test2", "X-Value2"))
+            .wrap(Condition::new(true, DefaultHeaders::new()))
+            .wrap(Compat::new(Logger::default()));
+
+        #[cfg(feature = "__compress")]
+        {
+            let _ = App::new().wrap(Compress::default()).wrap(Logger::default());
+            // let _ = App::new().wrap(Logger::default()).wrap(Compress::default());
+            let _ = App::new().wrap(Compat::new(Compress::default()));
+            let _ = App::new().wrap(Condition::new(true, Compat::new(Compress::default())));
+        }
+    }
+}

--- a/src/middleware/mod.rs
+++ b/src/middleware/mod.rs
@@ -53,7 +53,7 @@ mod tests {
         #[cfg(feature = "__compress")]
         {
             let _ = App::new().wrap(Compress::default()).wrap(Logger::default());
-            // let _ = App::new().wrap(Logger::default()).wrap(Compress::default());
+            let _ = App::new().wrap(Logger::default()).wrap(Compress::default());
             let _ = App::new().wrap(Compat::new(Compress::default()));
             let _ = App::new().wrap(Condition::new(true, Compat::new(Compress::default())));
         }

